lib/jsdom/living/validity/ValidityState-impl.js

```diff
@@ -0,0 +1,66 @@
+"use strict";
+
+exports.implementation = class ValidityStateImpl {
+  get badInput() {
+    return this.hasErrors("badInput");
+  }
+
+  get customError() {
+    return this.hasErrors("customError");
+  }
+
+  get patternMismatch() {
+    return this.hasErrors("patternMismatch");
+  }
+
+  get rangeOverflow() {
+    return this.hasErrors("rangeOverflow");
+  }
+
+  get rangeUnderflow() {
+    return this.hasErrors("rangeUnderflow");
+  }
+
+  get stepMismatch() {
+    return this.hasErrors("stepMismatch");
+  }
+
+  get tooLong() {
+    return this.hasErrors("tooLong");
+  }
+
+  get tooShort() {
+    return this.hasErrors("tooShort");
+  }
+
+  get typeMismatch() {
+    return this.hasErrors("typeMismatch");
+  }
+
+  get valueMissing() {
+    return this.hasErrors("valueMissing");
+  }
+
+  hasErrors(method) {
```

Comment:
Methods private to IDL and not exposed should be prefixed with a single underscore.